pkg/gohai/filesystem/filesystem_nix_test.go

```diff
@@ -3,145 +3,356 @@
+			for _, mount := range inputMounts {
+				// We only care about excluding specific FS types that we are doing the
+				// test for and that are marked as excluded
+				if mount.FSType == tc.FSType {
+					if !tc.Included {
+						continue
+					}
+				}
+
+				expectedMount := MountInfo{
+					Name:      mount.Source,
+					MountedOn: mount.Mountpoint,
+					SizeKB:    1,
+				}
+
+				expectedMounts = append(expectedMounts, expectedMount)
+			}
+
+			mounts, err := getFileSystemInfoWithMounts(inputMounts, mockFSInfo)
+			require.NoError(t, err)
+
+			require.Equal(t, len(expectedMounts), len(mounts))
+			assert.ElementsMatch(t, mounts, expectedMounts)
+		})
+	}
 }
 
-func TestOldMacosDf(t *testing.T) {
-	// from https://apple.stackexchange.com/questions/263437/df-hide-ifree-iused-512-blocks-customize-column-format-dont-show-inode-info
-	withDfCommand(t, "sh", "-c", `
-		echo 'Filesystem                                 1K-blocks       Used Available Capacity  iused    ifree %iused  Mounted on';
-		echo '/dev/disk0s2                               975093952  719904648 254677304    74% 90052079 31834663   74%   /';
-		echo 'devfs                                            368        368         0   100%      637        0  100%   /dev';
-		echo 'map -hosts                                         0          0         0   100%        0        0  100%   /net';
-		echo 'map -static                                        0          0         0   100%        0        0  100%   /Volumes/Large';
-	`)
-
-	out, err := getFileSystemInfo()
-	require.NoError(t, err)
-	require.Equal(t, []interface{}{
-		map[string]string{"kb_size": "975093952", "mounted_on": "/", "name": "/dev/disk0s2"},
-		map[string]string{"kb_size": "368", "mounted_on": "/dev", "name": "devfs"},
-		map[string]string{"kb_size": "0", "mounted_on": "/net", "name": "map -hosts"},
-		map[string]string{"kb_size": "0", "mounted_on": "/Volumes/Large", "name": "map -static"},
-	}, out)
+func TestNixMissingMountValues(t *testing.T) {
+	mockFSInfo := newMockFSInfo()
+
+	var testCases = []struct {
+		Desc           string
+		InputMounts    []*mountinfo.Info
+		ExpectedMounts []MountInfo
+	}{
+		{
+			"MissingSize",
+			[]*mountinfo.Info{newTestInputMountinfo("Normal1")},
+			[]MountInfo{newTestOutputMountInfo("Normal1")},
+		},
+		{
+			"MissingMountName",
+			[]*mountinfo.Info{
+				newTestInputMountinfo("Normal1"),
+				{Source: "", FSType: "foo", Mountpoint: "Bad1"},
+				newTestInputMountinfo("Normal2"),
+				{Source: "none", FSType: "foo", Mountpoint: "Bad2"},
+				newTestInputMountinfo("Normal3"),
+			},
+			[]MountInfo{
+				newTestOutputMountInfo("Normal1"),
+				newTestOutputMountInfo("Normal2"),
+				newTestOutputMountInfo("Normal3"),
+			},
+		},
+		{
+			"MissingMountPoint",
+			[]*mountinfo.Info{
+				newTestInputMountinfo("Normal1"),
+				{Source: "Bad1", FSType: "foo", Mountpoint: ""},
+				newTestInputMountinfo("Normal2"),
+				{Source: "Bad2", FSType: "foo", Mountpoint: ""},
```

Comment:
The 2 bad mount seems to be the same. Is there a reason to add the same test case twice ?

Reply:
Done in c211175c04c0b1b489eb2318558e4fc9bf1d3578